scipy/optimize/_qap.py

```diff
@@ -0,0 +1,709 @@
+
+    Graph matching tries to *maximize* the same objective function.
+    This algorithm can be thought of as finding the alignment of the
+    nodes of two graphs that minimizes the number of induced edge
+    disagreements, or, in the case of weighted graphs, the sum of squared
+    differences of edge weight disagreements.
+
+    Note that the quadratic assignment problem is NP-hard, is not
+    known to be solvable in polynomial time, and is computationally
+    intractable. Therefore, the results given are approximations,
+    not guaranteed to be exact solutions.
+
+
+    Parameters
+    ----------
+    A : 2d-array, square, non-negative
+        The square adjacency matrix :math:`A` in the objective function above.
+
+    B : 2d-array, square, non-negative
+        The square adjacency matrix :math:`B` in the objective function above.
+
+    method :  str in {'faq', '2opt'}
+        The algorithm used to solve the problem.
+        :ref:`'faq' <optimize.qap-faq>` (default) and
+        :ref:`'2opt' <optimize.qap-2opt>` are available.
+
+    options : dict, optional
+        A dictionary of solver options. All solvers support the following:
+
+        partial_match : 2d-array of integers, optional, (default = None)
+            Allows the user to fix part of the matching between the two
+            adjacency matrices. In the literature, a partial match is also
+            known as a "seed" [2]_.
+
+            Each row of `partial_match` specifies the indices of a pair of
+            corresponding nodes, that is, node ``partial_match[i, 0]`` is
+            matched to node ``partial_match[i, 1]``. Accordingly,
+            ``partial_match`` is an array of size ``(m , 2)``, where ``m`` is
+            less than the number of nodes.
+
+        maximize : bool (default = False)
+            Setting `maximize` to ``True`` solves the Graph Matching Problem
+            (GMP) rather than the Quadratic Assingnment Problem (QAP).
+
+        rng : {None, int, `~np.random.RandomState`, `~np.random.Generator`}
+            This parameter defines the object to use for drawing random
+            variates.
+            If `rng` is ``None`` the `~np.random.RandomState` singleton is
+            used.
+            If `rng` is an int, a new ``RandomState`` instance is used,
+            seeded with `rng`.
+            If `rng` is already a ``RandomState`` or ``Generator``
+            instance, then that object is used.
+            Default is None.
+
+    Returns
+    -------
+    res : OptimizeResult
+        A :class:`scipy.optimize.OptimizeResult` containing the following
+        fields.
+
+        col_ind : 1-D array
+            An array of column indices corresponding with the best
+            permutation of the nodes of `B` found.
+        score : float
+            The corresponding value of the objective function.
+        nit : int
+            The number of Franke-Wolfe iterations performed during
```

Comment:
Presumably these are not "Franke-Wolfe iterations" If 2-opt is used, just "iterations"?

Reply:
OK fixed that, too; see most recent commit. For the rest, please add in your PR.